packages/flutter_tools/lib/src/commands/clean.dart

```diff
@@ -28,41 +28,38 @@ class CleanCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    final FlutterProject flutterProject = FlutterProject.current();
     final Directory buildDir = fs.directory(getBuildDirectory());
+    _deleteFile(buildDir);
 
-    printStatus("Deleting '${buildDir.path}${fs.path.separator}'.");
-    if (buildDir.existsSync()) {
-      try {
-        buildDir.deleteSync(recursive: true);
-      } on FileSystemException catch (error) {
-        if (platform.isWindows) {
-          _windowsDeleteFailure(buildDir.path);
-        }
-        throwToolExit(error.toString());
-      }
-    }
+    final FlutterProject flutterProject = FlutterProject.current();
+    _deleteFile(flutterProject.dartTool);
 
-    printStatus("Deleting '${flutterProject.dartTool.path}${fs.path.separator}'.");
-    if (flutterProject.dartTool.existsSync()) {
+    final Directory androidEphemeralDirectory = flutterProject.android.ephemeralDirectory;
+    _deleteFile(androidEphemeralDirectory);
+
+    final Directory iosEphemeralDirectory = flutterProject.ios.ephemeralDirectory;
+    _deleteFile(iosEphemeralDirectory);
+
+    return const FlutterCommandResult(ExitStatus.success);
+  }
+
+  void _deleteFile(FileSystemEntity file) {
+    final String path = file.path;
+    printStatus("Deleting '$path${fs.path.separator}'.");
+    if (file.existsSync()) {
```

Comment:
nit: flip the sense of the if and un-indent, like:
```dart
if (!file.existsSync()) {
  return;
}
try {
  ...
}
...
```
you might also move the `printStatus()` after checking that the object exists.